pkg/status/syncer.go

```diff
@@ -0,0 +1,85 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	bigtablev1 "bigtable-autoscaler.com/m/v2/api/v1"
+	"bigtable-autoscaler.com/m/v2/pkg/interfaces"
+	"github.com/go-logr/logr"
+	"golang.org/x/sync/errgroup"
+)
+
+const optimisticLockError = "the object has been modified; please apply your changes to the latest version and try again"
+const tickTime = 3 * time.Second
+
+type syncer struct {
+	ctx               context.Context
+	statusWriter      interfaces.WriterWrapper
+	autoscaler        *bigtablev1.BigtableAutoscaler
+	googleCloudClient interfaces.GoogleCloudClient
+	clusterID         string
+	log               logr.Logger
+}
+
+func NewSyncer(
+	ctx context.Context,
+	statusWriter interfaces.WriterWrapper,
+	autoscaler *bigtablev1.BigtableAutoscaler,
+	googleCloundClient interfaces.GoogleCloudClient, clusterID string, log logr.Logger,
+) (*syncer, error) {
+	if autoscaler.Status.CurrentCPUUtilization == nil {
+		var cpuUsage int32
+		autoscaler.Status.CurrentCPUUtilization = &cpuUsage
+	}
+
+	return &syncer{
+		ctx:               ctx,
+		statusWriter:      statusWriter,
+		autoscaler:        autoscaler,
+		googleCloudClient: googleCloundClient,
+		clusterID:         clusterID,
+		log:               log,
+	}, nil
+}
+
+func (s *syncer) SyncStatus() {
```

Comment:
`SyncStatus` seems misleading, it sounds like it is a one-time execution, while its actually a parallel looped routine beings started. I guess `Start()` would be a better name here.